src/ophyd_async/core/detector.py

```diff
@@ -0,0 +1,178 @@
+"""Module which defines abstract classes to work with detectors"""
+import asyncio
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import AsyncIterator, Dict, Optional, Sequence
+
+from bluesky.protocols import (
+    Asset,
+    Configurable,
+    Descriptor,
+    Readable,
+    Reading,
+    Stageable,
+    Triggerable,
+    WritesExternalAssets,
+)
+
+from .async_status import AsyncStatus
+from .device import Device
+from .signal import SignalR
+from .utils import merge_gathered_dicts
+
+
+class DetectorTrigger(Enum):
+    #: Detector generates internal trigger for given rate
+    internal = ()
+    #: Expect a series of arbitrary length trigger signals
+    edge_trigger = ()
+    #: Expect a series of constant width external gate signals
+    constant_gate = ()
+    #: Expect a series of variable width external gate signals
+    variable_gate = ()
+
+
+class DetectorControl(ABC):
+    @abstractmethod
+    def get_deadtime(self, exposure: float) -> float:
+        """For a given exposure, how long should the time between exposures be"""
+
+    @abstractmethod
+    async def arm(
+        self,
+        trigger: DetectorTrigger = DetectorTrigger.internal,
+        num: int = 0,
```

Comment:
change to optional int, to keep with convention of timeout being able to be None.

Reply:
Not bothering at the moment as this is an internal mechanism